src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -272,6 +281,13 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkCreateObject<T>>,
       options: SavedObjectsCreateOptions = {}
     ): Promise<SavedObjectsBulkResponse<T>> => {
+      // If objects contain workspace id and options.overwrite is true, it is a bulkCreate action.
+      const isUpdateWorkspaceMode =
+        objects.some((obj) => obj.id && obj.type === WORKSPACE_TYPE) && options.overwrite;
+      // Only OSD admin can bulkCreate workspace.
+      if (objects.some((obj) => obj.type === WORKSPACE_TYPE) && !isUpdateWorkspaceMode)
+        throw generateOSDAdminPermissionError();
```

Comment:
```suggestion
      // Objects with id in overwrite mode will be regarded as update
      const objectsToCreate: Array<SavedObjectsBulkCreateObject<T>> = options.overwrite ? objects.filter((obj) => !obj.id) : objects;
      // Only OSD admin can bulkCreate workspace.
      if (objectsWaitToCreate.some((obj) => obj.type === WORKSPACE_TYPE))
        throw generateOSDAdminPermissionError();
```
How do you like the implementation like this?

Reply:
Sure, the logic has been updated. Thanks.